### ISSUE TITLE
can not use credentials in folder

### ISSUE THREAD

**Issue by snadn** [jenkinsci/kubernetes-cd-plugin#19]
I use credentials in folder and get `ERROR: Failed to validate SSH credentials: null`.

if I change the credentials to global, it works well.

**Comment by ArieShout** [comment]
Do you mean if you store the SSH credentials in a folder (organized by Jenkins) and we cannot get the SSH credential?

**Comment by snadn** [comment]
@ArieShout yes,  https://plugins.jenkins.io/cloudbees-folder

**Comment by ArieShout** [comment]
The Folders plugin is used to organize jobs rather than the credentials. For credentials we can add them to different domain. I'm not sure if you mean that.

I tried different combinations of

* store the SSH credentials in global domain, or a subdomain
* create the project in the top level hierarchy, or in a folder

But I cannot reproduce the issue. Could you please provide more details about your setup, and paste the  stacktrace if any?

**Comment by snadn** [comment]
@ArieShout In my company, different teams use different folders, and the teams need to do permission isolation

**Comment by snadn** [comment]
my config is 
![image](https://user-images.githubusercontent.com/1269858/37824352-4045a680-2ec7-11e8-8b3d-784d24c96767.png)

and build log
```
Building on master in workspace /var/lib/jenkins/workspace/test/job
Starting Kubernetes deployment
ERROR: ERROR: null
java.lang.NullPointerException
	at com.microsoft.jenkins.azurecommons.remote.UsernameAuth.fromCredentials(UsernameAuth.java:37)
	at com.microsoft.jenkins.azurecommons.remote.SSHClient.<init>(SSHClient.java:53)
	at com.microsoft.jenkins.kubernetes.credentials.SSHCredentials$ClientWrapperFactoryImpl.fetchConfig(SSHCredentials.java:148)
	at com.microsoft.jenkins.kubernetes.credentials.SSHCredentials$ClientWrapperFactoryImpl.buildClient(SSHCredentials.java:139)
	at com.microsoft.jenkins.kubernetes.command.DeploymentCommand$1.call(DeploymentCommand.java:75)
	at com.microsoft.jenkins.kubernetes.command.DeploymentCommand$1.call(DeploymentCommand.java:66)
	at hudson.FilePath.act(FilePath.java:1096)
	at com.microsoft.jenkins.kubernetes.command.DeploymentCommand.execute(DeploymentCommand.java:66)
	at com.microsoft.jenkins.kubernetes.command.DeploymentCommand.execute(DeploymentCommand.java:45)
	at com.microsoft.jenkins.azurecommons.command.CommandService.runCommand(CommandService.java:88)
	at com.microsoft.jenkins.azurecommons.command.CommandService.execute(CommandService.java:96)
	at com.microsoft.jenkins.azurecommons.command.CommandService.executeCommands(CommandService.java:75)
	at com.microsoft.jenkins.azurecommons.command.BaseCommandContext.executeCommands(BaseCommandContext.java:77)
	at com.microsoft.jenkins.kubernetes.KubernetesDeploy.perform(KubernetesDeploy.java:42)
	at hudson.tasks.BuildStepCompatibilityLayer.perform(BuildStepCompatibilityLayer.java:81)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:744)
	at hudson.model.Build$BuildExecution.build(Build.java:206)
	at hudson.model.Build$BuildExecution.doRun(Build.java:163)
	at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:504)
	at hudson.model.Run.execute(Run.java:1727)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:97)
	at hudson.model.Executor.run(Executor.java:429)
ERROR: Kubernetes deployment ended with HasError
Finished: FAILURE
```

**Comment by ArieShout** [comment]
Hi @snadn , I can reproduce the issue now. I will check how we can fix this.
Thanks,

**Comment by ArieShout** [comment]
Fixed in release 0.2.0.

**Comment by snadn** [comment]
@ArieShout any more. expect the new kubeconfig can select credential in folder. only can select credential in global now.

**Comment by ArieShout** [comment]
Could you give some more details of your problem?

It works in my Jenkins instance. When we list the kubeconfig credentials and lookup the credentials in build, we take the enclosing folder into account, so it should work fine as the (deprecated) SSH credentials lookup.

I have a folder `k8s-test`, which contains a kubeconfig credential with ID `k8s-test`:

![image](https://user-images.githubusercontent.com/423695/38295790-d8138ce4-3821-11e8-96ae-7618d6781659.png)

In the same folder `k8s-test`, I configured a job named `k8s-test`, and in the job configuration, I can select the credentials configured in the folder:

![image](https://user-images.githubusercontent.com/423695/38295835-0763b8d4-3822-11e8-949e-2cf803d6bb28.png)

And when it goes to deployment, it can get the kubeconfig correctly.

**Comment by snadn** [comment]
I'm not talking about the kubeconfig type of credential , I mean when creating kubeconfig, I can't choose ssh credential in the folder

**Comment by ArieShout** [comment]
I see. It seems this is not trivial to fix. I will log another issue to track this.

**Comment by snadn** [comment]
👌

**Comment by ArieShout** [comment]
@snadn This is now fixed in release 0.2.1
